### ISSUE TITLE
Metadata local-hostname should have FQDN instead of short hostname

### ISSUE THREAD

**Issue by remibergsma** [MissionCriticalCloud/cosmic#450]
Creating issue from e-mail received to keep track of it.

Hi guys,
    
For a while now, we’ve seen cloud-init behaving oddly when starting up new centos6 templates, specifically setting the domain of the machine to be .localdomain. Then, this needs to be manually fixed for people before they can move onto their configuration. We spent some time on Friday really digging into this, and eventually in the code we found that in the metadata of the machine, the current ‘standard’ for cloud-init is for the domain to be appended to the machine name, from the network domain. At the moment however, cloudstack/cosmic is only providing the short name of the machine, and then the default of ‘localdomain’ is appended.
    
```
https://github.com/number5/cloud-init/blob/master/cloudinit/sources/__init__.py#L182
```
        
In short, on the router vms, the file local-hostname in `/var/www/html/metadata/$instance_IP/local-hostname` needs to have the network domain appended to it. An example is vm app26, which is in a network of bla.local, at the moment the file only has ‘app26’ when it needs to have ‘app26.bla.local’
    
I hope this makes sense, please let me know if I can provide any further information. Could this be added to the backlog to be fixed in a new version of cosmic please?

**Comment by remibergsma** [comment]
Fixed by #452
